src/fides/api/service/connectors/scylla_connector.py

```diff
@@ -0,0 +1,97 @@
+from typing import Any, Dict, List, Optional
+
+import cassandra
+from cassandra.auth import PlainTextAuthProvider
+from cassandra.cluster import Cluster
+from loguru import logger
+
+from fides.api.common_exceptions import ConnectionException
+from fides.api.graph.execution import ExecutionNode
+from fides.api.models.connectionconfig import ConnectionTestStatus
+from fides.api.models.policy import Policy
+from fides.api.models.privacy_request import PrivacyRequest, RequestTask
+from fides.api.schemas.connection_configuration.connection_secrets_scylla import (
+    ScyllaSchema,
+)
+from fides.api.service.connectors.base_connector import BaseConnector
+from fides.api.service.connectors.query_config import QueryConfig
+from fides.api.util.collection_util import Row
+
+
+class ScyllaConnector(BaseConnector):
+    """Scylla Connector"""
+
+    def build_uri(self) -> str:
+        """
+        Builds URI
+        """
+
+    def create_client(self) -> Cluster:
+        """Returns a Scylla cluster"""
+
+        config = ScyllaSchema(**self.configuration.secrets or {})
+
+        auth_provider = PlainTextAuthProvider(
+            username=config.username, password=config.password
+        )
+        cluster = Cluster([config.host], port=config.port, auth_provider=auth_provider)
+        return cluster
+
+    def query_config(self, node: ExecutionNode) -> QueryConfig[Any]:
+        pass
+
+    def test_connection(self) -> Optional[ConnectionTestStatus]:
+        """
+        Connects to the scylla database and issues a trivial query
+        """
+        logger.info("Starting test connection to {}", self.configuration.key)
+        config = ScyllaSchema(**self.configuration.secrets or {})
+        cluster = self.client()
+
+        try:
+            with cluster.connect(
+                config.keyspace if config.keyspace else None
+            ) as client:
+                client.execute("select now() from system.local")
+        except cassandra.cluster.NoHostAvailable as exc:
```

Comment:
very thorough exception handling here! taking some notes on the test/error cases to improve my own handling in the s3 connector - seems worth the effort given that this is the primary diagnostic hook/tool for connection debugging 👍